src/platform/testing/unit/helpers.js

```diff
@@ -99,6 +99,39 @@ export function mockFetch(returnVal, shouldResolve = true) {
     );
 }
 
+export function setFetchJSONResponse(stub, data = null) {
```

Comment:
This PR looks a little scarier than it really is. The bulk of the work is taking existing helper functions and moving them into this common unit test helpers file. And then using these new common helpers in the test files.